### PR TITLE
Add multi-factor authentication guide page.

### PR DESCRIPTION
**Pull request by ecnelises**

### CODE REVIEW

File: using-multifactor-authentication.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+QR-code. Be cautious to type the _key_ correctly and choose _time based_
+option, and click _save_. You can see a 6-digit code changing with time (30
+seconds) in your app. Type the code and click _enable_.
+
+If the code is correct and QR-code not expired, you can see a list of recovery
+codes. Please keep them safe. Any of these codes can be used in place of digit
+code generated from authenticator, but every one can only be used once. So if
+your phone is missing or you unexpectedly deleted the account in the
+authenticator, you can use recovery code to disable multi-factor authentication
+first and re-generate a new key.
```
Comment (sonalkr132) [inline using-multifactor-authentication.md]: @ecnelises I think a numbered list of steps to take with appropriate screenshots would work better here. Something similar to https://docs.microsoft.com/en-us/azure/active-directory/user-help/microsoft-authenticator-app-how-to#add-a-work-or-school-account

File: using-multifactor-authentication.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+- _disabled_, setting to this level is equivalent to disable your MFA.
+- _login only_, only login from browser or command line requires digit code.
+- _login and write_, login, pushing gems or changing gem ownership requires
+digit code.
+
+All level change requires authentication from digit code, or recovery code.
+To clear your multifactor authentication settings, choose _disabled_ and
+continue.
+
+## Using multi-factor authentication in command line
```
Comment (sonalkr132) [inline using-multifactor-authentication.md]: Lets use a seperate page for this

File: using-multifactor-authentication.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    Email:   gem_author@example
+    Password:
+    
+    This command needs digit code for multifactor authentication.
+    Code:   111111
+    Signed in.
+
+There's a command line option for providing the digit code before execution.
+
+    $ gem signin --mfa 111111
```
Comment (sonalkr132) [inline using-multifactor-authentication.md]: have we used `mfa` as flag name? Please use `--otp`

File: using-multifactor-authentication.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+Likewise, `gem push` and `gem owner` requires the code if your authentication
+level is set to _login and write_.
+
+    $ gem push hello-0.0.1.gem
+    Pushing gem to https://rubygems.org...
+    This command needs digit code for multifactor authentication.
+    Code:   111111
+    Successfully registered gem: hello (0.0.1)
+
+If your input is incorrect, the operation would fail.
```
Comment (sonalkr132) [inline using-multifactor-authentication.md]: Just leave a not about otp expiring in 30 sec, we don't need to show example of incorrect input.

File: using-multifactor-authentication.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+    $ gem owner hello --add gem_author2@example
+    This command needs digit code for multifactor authentication.
+    Code:   222222
+    Adding gem_author2@example: You have enabled multifactor authentication but your request doesn't have the correct OTP code. Please check it and retry.
+
+One thing should be noted out is that signing in will not keep mfa stored. So
+if you are not logged in, some operations may ask you twice for MFA code.
```
Comment (sonalkr132) [inline using-multifactor-authentication.md]: Please explain here which exact operations will always ask otp and why did we choose not to go with a session based approach.

Comment (ecnelises) [reply using-multifactor-authentication.md]: Now we just keep otp in memory and won't ask for otp twice any more.

**Comment by sonalkr132** [comment]
See #221

**Pull request closed without merging**